src/main.cpp

```diff
@@ -3447,6 +3451,23 @@ bool static DisconnectTip(CValidationState& state, const CChainParams& chainpara
         }
     }
 
+    if(pblocktree->GetProposalIndex(vecProposal)){
+
+         for(unsigned int i = 0; i < vecProposal.size(); i++) {
+             propsal = vecProposal[i];
+
+             if(propsal.blockhash == pindexDelete->GetBlockHash()) {
```

Comment:
blockhash points to the block where the proposal was accepted, so this condition is only triggered if the proposal was accepted on the block to disconnect